docs/extending.txt

```diff
@@ -549,3 +549,44 @@ arguments.  Example use:
         endswith=endswith) + t
 
     site.template_hooks['page_header'].append(greeting, True, u'Nikola Tesla', endswith=u'!')
+
+Shortcodes
+==========
+
+Some (hopefully all) markup compilers support shortcodes in these forms::
```

Comment:
About ipynb ... this runs over everything pretty dumbly. In fact, now that you mention it, adding it to the rest compiler would break the docs, now that they mention shortcodes ;-)